meson.build

```diff
@@ -0,0 +1,1974 @@
+# -*- mode: meson -*-
+
+project('systemd', 'c',
+        version : '233',
+        license : 'LGPLv2+',
+        default_options: [
+          'c_std=gnu99',
+          'prefix=/usr',
+          'sysconfdir=/etc',
+          'localstatedir=/var',
+          ],
+        meson_version : '>= 0.39.1',
+       )
+
+# We need the same data in three different formats, ugh!
+# Also, for hysterical reasons, we use different variable
+# names, sometimes. Not all variables are included in every
+# set. Ugh, ugh, ugh!
+conf = configuration_data()
+conf.set_quoted('PACKAGE_STRING',  meson.project_name() + ' ' + meson.project_version())
+conf.set_quoted('PACKAGE_VERSION', meson.project_version())
+
+substs = configuration_data()
+substs.set('PACKAGE_URL',          'https://www.freedesktop.org/wiki/Software/systemd')
+substs.set('PACKAGE_VERSION',      meson.project_version())
+
+m4_defines = []
+
+#####################################################################
+
+if get_option('split-usr')
+  conf.set('HAVE_SPLIT_USR', 1)
+  rootprefixdir = '/'
+else
+  rootprefixdir = '/usr'
+endif
+
+sysvinit_path = get_option('sysvinit-path')
+sysvrcnd_path = get_option('sysvrcnd-path')
+if sysvinit_path != '' or sysvrcnd_path != ''
+  conf.set('HAVE_SYSV_COMPAT', 1,
+           description : 'SysV init scripts and rcN.d links are supported')
+endif
+
+# join_paths ignore the preceding arguments if an absolute component is
+# encountered, so this should canonicalize various paths when they are
+# absolute or relative.
+prefixdir = get_option('prefix')
+if not prefixdir.startswith('/')
+  error('Prefix is not absolute: "@0@"'.format(prefixdir))
+endif
+bindir = join_paths(prefixdir, get_option('bindir'))
+libdir = join_paths(prefixdir, get_option('libdir'))
+sysconfdir = join_paths(prefixdir, get_option('sysconfdir'))
+includedir = join_paths(prefixdir, get_option('includedir'))
+datadir = join_paths(prefixdir, get_option('datadir'))
+localstatedir = join_paths('/', get_option('localstatedir'))
+
+rootbindir = join_paths(rootprefixdir, 'bin')
+rootlibdir = join_paths(rootprefixdir, libdir.split('/')[-1])
```

Comment:
That doesn't work properly on Debian, where libdir is typically lib/$(architecture). See https://github.com/systemd/systemd/pull/5704#issuecomment-292951931, where pam_systemd.so ends up as
```/x86_64-linux-gnu/security/pam_systemd.so```